src/Raven.Server/Utils/ThrottledManualResetEventSlim.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Threading;
+using Sparrow.Threading;
+
+namespace Raven.Server.Utils
+{
+    public class ThrottledManualResetEventSlim : IDisposable
+    {
+        private readonly ManualResetEventSlim _mre;
+        private readonly MultipleUseFlag _setCalled = new MultipleUseFlag();
+        private Timer _timer;
+        private TimeSpan? _throttlingTimeInterval;
+        private bool _throttlingEnabled;
+
+        public ThrottledManualResetEventSlim(TimeSpan? throttlingTime, bool initialState = false)
+        {
+            _mre = new ManualResetEventSlim(initialState);
+
+            if (throttlingTime != null)
+                EnableThrottling(throttlingTime.Value);
```

Comment:
TODO: we spawn a timer immediately after creating the instance - while the index might be disabled. The timer should be active only if an index is enabled